app/src/free/java/org/mozilla/fenix/components/metrics/AdvertisingIDImpl.kt

```diff
@@ -0,0 +1,13 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.components.metrics
+
+import android.content.Context
+
+interface AdvertisingIDImpl : AdvertisingID {
```

Comment:
Should this be a class instead of an interface?

Reply:
Rolledback to an interface because otherwise detekt was complaining of an abstract class containing only abstract method (see [UnncessaryAbstractClass](https://arturbosch.github.io/detekt/style.html#unnecessaryabstractclass)). Tell me if you think it would be better off keeping a class and ignoring the warning using an annotation.